interop-cats/jvm/src/test/scala/zio/interop/Fs2ZioSpec.scala

```diff
@@ -4,113 +4,98 @@ package interop
-        released <- Promise.make[Nothing, Unit]
-        fail     <- Promise.make[Nothing, Unit]
-        _ <- Stream
-              .bracket(started.succeed(()).unit)(_ => released.succeed(()).unit)
-              .evalMap[Task, Unit] { _ =>
-                fail.await *> IO.fail(new Exception())
-              }
-              .compile
-              .drain
-              .fork
-
-        _ <- started.await
-        _ <- fail.succeed(())
-        _ <- released.await
-      } yield ()).timeout(10.seconds).provideLayer(ZEnv.live)
-    } must beSome(())
-
-  def bracketTerminate =
-    unsafeRun {
-      (for {
-        started   <- Promise.make[Nothing, Unit]
-        released  <- Promise.make[Nothing, Unit]
-        terminate <- Promise.make[Nothing, Unit]
-        _ <- Stream
-              .bracket(started.succeed(()).unit)(_ => released.succeed(()).unit)
-              .evalMap[Task, Unit] { _ =>
-                terminate.await *> IO.die(new Exception())
-              }
-              .compile
-              .drain
-              .fork
-
-        _ <- started.await
-        _ <- terminate.succeed(())
-        _ <- released.await
-      } yield ()).timeout(10.seconds).provideLayer(ZEnv.live)
-    } must beSome(())
-
-  def bracketInterrupt =
-    unsafeRun {
-      (for {
-        started  <- Promise.make[Nothing, Unit]
-        released <- Promise.make[Nothing, Unit]
-        f <- Stream
-              .bracket(IO.unit)(_ => released.succeed(()).unit)
-              .evalMap[Task, Unit](_ => started.succeed(()) *> IO.never)
-              .compile
-              .drain
-              .fork
-
-        _ <- started.await
-        _ <- f.interrupt
-        _ <- released.await
-      } yield ()).timeout(10.seconds).provideLayer(ZEnv.live)
-    } must beSome(())
+  def bracketFail: ZIO[Any, Nothing, TestResult] =
+    for {
+      started  <- Promise.make[Nothing, Unit]
+      released <- Promise.make[Nothing, Unit]
+      fail     <- Promise.make[Nothing, Unit]
+      _ <- Stream
+            .bracket(started.succeed(()).unit)(_ => released.succeed(()).unit)
+            .evalMap[Task, Unit] { _ =>
+              fail.await *> IO.fail(new Exception())
+            }
+            .compile
+            .drain
+            .fork
+
+      _ <- started.await
+      _ <- fail.succeed(())
+      _ <- released.await
+    } yield assert(())(anything)
+
+  def bracketTerminate: ZIO[Any, Nothing, TestResult] =
+    for {
+      started   <- Promise.make[Nothing, Unit]
+      released  <- Promise.make[Nothing, Unit]
+      terminate <- Promise.make[Nothing, Unit]
+      _ <- Stream
+            .bracket(started.succeed(()).unit)(_ => released.succeed(()).unit)
+            .evalMap[Task, Unit] { _ =>
+              terminate.await *> IO.die(new Exception())
+            }
+            .compile
+            .drain
+            .fork
+
+      _ <- started.await
+      _ <- terminate.succeed(())
+      _ <- released.await
+    } yield assert(())(anything)
```

Comment:
is this assertion make any sense? shouldn't we test results?

Reply:
@molowny

Replaced `assert(())(anything)` with `assertCompletes` as there is already a nice combinator to achieve such check.

The idea of this test is to check if things "work when fiber is interrupted". If there is an error somewhere, it will be encoded in ZIO and found by zio-test. (At least that is my understanding.)